packages/web-app-files/src/mixins/fileActions.js

```diff
@@ -135,7 +136,16 @@ export default {
       })
     },
 
+    // TODO: Make user-configurable what is a defaultAction for a filetype/mimetype
+    // returns the _first_ action from actions array which we now construct from
+    // available mime-types coming from the app-provider and existing actions
     $_fileActions_triggerDefaultAction(resource) {
+      const availableExternalAppActions = this.$_fileActions_loadApps(resource)
+
+      for (const action of availableExternalAppActions) {
+        action.handler = () => this.$_fileActions_openLink(action.name, resource.fileId)
```

Comment:
If we move the app name from the route (`ocis.owncloud.test/#/external/Collabora/MTI4....) in a query parameter (ocis.owncloud.test/#/external/MTI4...?app=Collabora), we could skip it. 

Calling `/app/open` without a `app_name` query parameter already opens it in the default application.
Additionally we also could add information on the `/app/list` endpoint, which application is the default.